lib/_http_client.js

```diff
@@ -24,6 +24,18 @@ function ClientRequest(options, cb) {
     if (!options.hostname) {
       throw new Error('Unable to determine the domain name');
     }
+  } else if (options instanceof url.URL) {
```

Comment:
If I understand correctly this makes using the new URL api and using additional options(like `agent`) mutual exclusive?